### PR TITLE
[Communication] - Identity/TURN - Update TURN SDK Version and convert TURN Urls response to array

### PR DESCRIPTION
**Pull request by apattath**
The following changes are present in this PR (all are based on comments on my original SDK check-in here: https://github.com/Azure/azure-sdk-for-net/pull/18865). 

1. Update SDK to use TURN preview version
2. Update SDK to handle Urls field as an array in TURN response (The corresponding swagger changes can be found here https://github.com/Azure/azure-rest-api-specs/pull/13157)
3. Add Managed Identity TURN API tests.
4. Update code snippets and ReadMe files to match above changes.

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/communication/Azure.Communication.Identity/README.md
Comment (ajpeacock0) [inline sdk/communication/Azure.Communication.Identity/README.md:174]: Is this needed? Also the `version=1.0.0-beta.4` is wrong here

**Review by ajpeacock0 — APPROVED** [review]

File: sdk/communication/Azure.Communication.Identity/README.md
Comment (apattath) [reply sdk/communication/Azure.Communication.Identity/README.md:174]: I wasn't sure what's the best way for this readme to be self-documenting. I wanted to link to the latest released SDK version, but that can't be determined until the PR is merged and pipeline is run.

File: sdk/communication/Azure.Communication.Identity/src/CommunicationIdentityClient.cs
Comment (apattath) [inline sdk/communication/Azure.Communication.Identity/src/CommunicationIdentityClient.cs:85]: Unfortunately the change to unify both Identity and TURN clients didn't make it into the PROD deployment, so I couldn't fix this hacky version of using two separate clients here.

File: sdk/communication/Azure.Communication.Identity/README.md
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Identity/README.md:14]: We do not need to specify that is private here.

File: sdk/communication/Azure.Communication.Identity/README.md
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Identity/README.md:17]: Why do they need to explicitly import Common?

File: sdk/communication/Azure.Communication.Identity/README.md
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Identity/README.md:174]: I like that you are trying to have full documentation here. It might be better to explain which NuGet feed should be added.

File: sdk/communication/Azure.Communication.Identity/src/CommunicationIdentityClient.cs
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Identity/src/CommunicationIdentityClient.cs:85]: When does it reach prod?

File: sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs:68]: 👍🏻

File: sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs:71]: Now that it is consistent, maybe better to extract this client creation into a method and use it across

File: sdk/communication/Azure.Communication.Identity/src/CommunicationIdentityClient.cs
Comment (ajpeacock0) [reply sdk/communication/Azure.Communication.Identity/src/CommunicationIdentityClient.cs:85]: It reached Prod yesterday

**Review by RezaJooyandeh — APPROVED** [review]
I left some comments, but none of them need to block this PR. Please make sure to address them in the one though as it is going to the feature branch.

File: sdk/communication/Azure.Communication.Identity/README.md
Comment (apattath) [reply sdk/communication/Azure.Communication.Identity/README.md:17]: I guess they don't need to. Nuget should pull any dependent packages. I will remove it.

File: sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs
Comment (apattath) [reply sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs:71]: Done.

**Pull request merged**
